BoxContentSDK/BoxContentSDK/Clients/BOXContentClient.m

```diff
@@ -486,6 +493,27 @@ + (NSString *)APIBaseURL
     return staticAPIBaseURL;
 }
 
+//NOTE: Right now we assume that version-portion of the URL string is specified by the last component in the URL path if it also starts with a number
++ (NSString *)APIBaseURLWithoutVersion
+{
+    if (staticAPIBaseURLWithoutVersion.length == 0) {
+        NSString *versionedURL = [BOXContentClient APIBaseURL];
+        NSString *unversionedURL = nil;
+        NSURL *url = [NSURL URLWithString:versionedURL];
+        NSArray<NSString *> *components = [url pathComponents];
+        NSString *lastComponent = [components lastObject];
+        if( [lastComponent length] && isnumber([lastComponent characterAtIndex:0])) {
+            NSString *removeString = [NSString stringWithFormat:@"/%@", lastComponent];
+            unversionedURL = [versionedURL stringByReplacingOccurrencesOfString:removeString withString:@""];
+        }
```

Comment:
You can do this more simply actually. NSString has 'pathComponents' on it directly, as well as 'lastPathComponent'. And once you've verified that our last path component is a version, you can simply do 'stringByDeletingLastPathComponent'. That should work, right?